frontend/views/containers/proposals/ProposalsWidget.vue

```diff
@@ -53,6 +53,10 @@ export default ({
     ProposalItem,
     ButtonDropdownMenu
   },
+  created () {
+    // check if there are expired proposals to archive
+    this.checkExpiredProposals()
```

Comment:
This does a single check once when then ProposalsWidget is loaded. But what if we are already loaded and the proposal expires while we are looking at the page?

We want to make sure:

1. The user cannot vote on a proposal that has actually expired. So when they vote the check should happen there.
2. That the check is performed periodically, say every 3 minutes or so. This might be a useful candidate to combine with your periodic notifications code perhaps

Reply:
@taoeffect Agree with below

> This might be a useful candidate to combine with your periodic notifications code perhaps

As you might remember, we currently have a periodic notification logic that checks for expiring proposals that recurs every 15mins. Since we are dealing with a very similar topic(`expiring` and `expired` proposals) here, I decided to make additions to that existing logic that it checks both 'expired' and 'expiring' proposals in it and then reduced the recurrence period down to `5min` from the previous `15min`. Hope it sounds good.

Reply:
@SebinSong That sounds like a great idea!  👍